constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/helper/widget/Carousel.java

```diff
@@ -294,9 +312,36 @@ private void updateItems() {
             View view = mList.get(i);
             int index = mIndex + i - startIndex;
             if (index < 0) {
-                updateViewVisibility(view, emptyViewBehavior);
+                if (infiniteCarousel) {
```

Comment:
instead of this -- would it be simpler to check if infiniteCarousel is true and then compute index as the modulo of the count? then the bulk of the code would stay the same

Reply:
Great, refactoring done.

Reply:
Great, refactoring done.